modules/dataquery/jsx/react.filterBuilder.js

```diff
@@ -568,13 +569,78 @@ class FilterGroup extends Component {
 class FilterBuilder extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      showModalCSV: false,
+    };
+    this.openModalCSV = this.openModalCSV.bind(this);
+    this.closeModalCSV = this.closeModalCSV.bind(this);
+    this.defineCSVCandidates = this.defineCSVCandidates.bind(this);
+  }
+
+  openModalCSV(e) {
+    e.preventDefault();
+    this.setState({showModalCSV: true});
+  }
+
+  closeModalCSV() {
+    this.setState({showModalCSV: false});
+  }
+
+  defineCSVCandidates(type, data) {
+    let session = [];
+    let children = [];
+    for (const key in data) {
+      if (data.hasOwnProperty(key)) {
+        const value = data[key][0];
+        children.push({
+          field: type,
+          fieldType: 'varchar(255)',
+          instrument: 'demographics',
+          operator: 'equal',
+          session: [value],
+          type: 'rule',
+          value: value,
+          visit: 'All'
+        });
+        session.push(value);
+      }
+    }
+    const results = {
+      session: session,
+      children: children,
+    };
+    fetch(
+      window.location.origin
+      + '/dataquery/View/csv',
+      {
+        credentials: 'same-origin',
+        method: 'PUT',
```

Comment:
We talked about this and you told me you had to use a PUT request here instead of a POST since apparently you were exceeding the max POST request size in some cases. I still think a POST request should be used and we should increase the max POST request size configured in `php.ini`.

Reply:
@nicolasbrossard why do you think it should be a POST?

Reply:
I came across this:

https://stackoverflow.com/questions/107390/whats-the-difference-between-a-post-and-a-put-http-request

I think our use case best fits the description of what a POST request is.

Reply:
When I read these links:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT

I'm sort of at loss on if a POST is even suitable.
> A POST request is typically sent via an HTML form and results in **a change on the server**.

> The difference between PUT and POST is that PUT is idempotent: calling it once or several times successively has the same effect (that is no side effect), whereas successive identical POST requests may have additional effects, akin to placing an order several times.

Anyway hopefully it won't matter when I switch to a POST if people just modify the php config file.